### ISSUE TITLE
Initial date bug

### ISSUE THREAD

**Issue by 3nematix** [probablykasper/date-picker-svelte#41]
When providing max date and not passing any date to keep the placeholder visible when u open the date picker the date will always show 2003 year and no days untill you select a month or another year. 

code to reproduce the bug
```svelte
<script>
    import { DateInput } from 'date-picker-svelte'

    let date;
    let date_string;
    let is_date_valid;
    let max_date = new Date();

    // Atleast 14 years old
    max_date.setDate(max_date.getDate() - 5110);

    $: if (date && is_date_valid) {
        date.setHours(12, 0, 0, 0)
        date.toISOString()
        date_string = date.toISOString().slice(0, 10);
    }
</script>

<input bind:value={date_string} type="date" class="hidden" id="id_birth_date" name="birth_date">
<DateInput bind:value={date} bind:valid={is_date_valid} max={max_date} format="yyyy-MM-dd" placeholder="Select a date"/>
```

**Comment by 3nematix** [comment]
Another one bug is that you can't even select any year before 2003 this works like a min date

https://user-images.githubusercontent.com/46891372/210883269-2158b0e1-ae51-4372-ba2e-3bbb85a60e31.mp4

**Comment by probablykasper** [comment]
Fixed in v2.4.2. If today is outside min/max range, It now shows the date closest to today
